pkg/scheduler/scheduler.go

```diff
@@ -143,9 +158,24 @@ func NewScheduler(cfg Config, limits Limits, log log.Logger, registerer promethe
 	})
 
 	s.activeUsers = util.NewActiveUsersCleanupWithDefaultValues(s.cleanupMetricsForInactiveUser)
+	subservices := []services.Service{s.requestQueue, s.activeUsers}
 
-	var err error
-	s.subservices, err = services.NewManager(s.requestQueue, s.activeUsers)
+	// Init the ring only if the ring-based service discovery mode is used.
+	if cfg.ServiceDiscovery.Mode == schedulerdiscovery.ModeRing {
+		s.schedulerLifecycler, err = schedulerdiscovery.NewRingLifecycler(cfg.ServiceDiscovery.SchedulerRing, log, registerer)
+		if err != nil {
+			return nil, err
+		}
+
+		s.schedulerRing, err = schedulerdiscovery.NewRingClient(cfg.ServiceDiscovery.SchedulerRing, "query-scheduler", log, registerer)
```

Comment:
Query-scheduler doesn't need to *watch* scheduler ring. Lifecycler already implements HTTP handler to report ring status.

Reply:
Right, fixed!

Reply:
Done in https://github.com/grafana/mimir/pull/2957/commits/bfcf2655a521264904b0f393bebdc3c776b0e45c and https://github.com/grafana/mimir/pull/2957/commits/647d72864bc59d0c870d2b0693844e6f67c2227f.